### ISSUE TITLE
Updates to the auto-completion seem to happen randomly

### ISSUE THREAD

**Issue by ginsterbusch** [geany/geany#1044]
Ran into that issue again, after it was gone for ages (had that with 1.22 or 1.21): The auto-completion data is not being updated properly (PHP).

Update time is set to the default 250 ms, but it looks like as if this is completely ignored, as the updates seem to happen rather random (neither does a simple file reload help, nor the classic close file + open again; also, the usual back-then-fix, saving the file and then waiting a bit, doesnt work, too).
And yes, I changed the update time, too. Didn't help. Same random update behaviour.

System Info:
Geany Version: 1.27 (self-compiled)
Geany Plugin Version: 1.27 (dito)
System: Linux Mint 17.3 aka Ubuntu 14.04.4 LTS + XFCE 4.12 / Enablement Stack / x86_64

cu, w0lf.

**Comment by b4n** [comment]
If reloading the file doesn't help, I would guess it's caused by mis-parsing of the file rather than updates not triggering.  Could you paste a file triggering the issue?

Also, does it happen with all files, or specific ones? all filetypes, or only PHP?  Could you try and see if disabling plugins helps?

**Comment by ginsterbusch** [comment]
So far, its been happening with PHP, HTML and JS files. Not with text files (including Markdown, Wiki Markup etc.). Disabled all plugins, but still surfacing.

The files, with which I initially noticed the issue, are custom work for a client  (custom CSV import plugin for WooCommerce), thus, sadly no file dump allowed. But these are rather on the small side - eg. the main file is about 20kb with approx 650 lines of code and ~50ish variables in use.

When opening regular files of eg. WP includes (post.php, taxonomy.php and media.php), and testing a bit, there was no measurable delay, neither when creating new variables, functions and classes, nor when using existing ones. Said files are huge compared to above mentioned ones ([post.php](https://core.trac.wordpress.org/browser/tags/4.5.1/src/wp-includes/post.php): 200kb, ~6100 lines of code, [taxonomy.php](https://core.trac.wordpress.org/browser/tags/4.5.1/src/wp-includes/taxonomy.php): 166kb, ~4900 lines of code, and [media.php](https://core.trac.wordpress.org/browser/tags/4.5.1/src/wp-includes/media.php): 129kb, ~3800 lines of code).

cu, w0lf.

**Comment by codebrainz** [comment]
Is the random delay in how long it takes for the auto-completion popup list to show on the screen, or how long it takes for the symbols it shows to be up-to-date? If the latter, is there a similar issue with the symbols shown in the Symbols tab in the sidebar?

Maybe the files from your client have some wonky encoding issues? You could use Geany (Document->Set Encoding) or iconv or such to try and normalize the encoding to something typical like UTF-8 as a test. Just a guess.

**Comment by ginsterbusch** [comment]
> Maybe the files from your client have some wonky encoding issues? You could use Geany (Document->Set Encoding) or iconv or such to try and normalize the encoding to something typical like UTF-8 as a test. Just a guess.

That was actually the first thing I thought. Including improper access rights - alas, I created them myself, in Geany 1.26, with UTF-8 encoding, on the very same disk. Access rights are correct, too - or does 1.27 differ in this aspect from 1.26 and lower?

> or how long it takes for the symbols it shows to be up-to-date? If the latter, is there a similar issue with the symbols shown in the Symbols tab in the sidebar?

haven't really been watching that yet; I aint got my interface cluttered with sidebars all the time, but I will try and test it whenever the issue resurfaces. Is there a known bug with this then?

cu, w0lf.

**Comment by codebrainz** [comment]
>  or does 1.27 differ in this aspect from 1.26 and lower?

Not sure, I don't think anything that could cause this issue, but I'm not terribly familiar with the tags-related code.

> Is there a known bug with this then?

Don't think so.

Is there any interesting output in `Help->Debug Messages`?

**Comment by ginsterbusch** [comment]
Forgot adding the debug messages - only regular INFO messages, expect two about Glib:


```
20:55:14: GLib CRITICAL	: Source ID 2738 was not found when attempting to remove it
20:56:16: GLib CRITICAL	: Source ID 3163 was not found when attempting to remove it
```

cu, w0lf.

**Comment by codebrainz** [comment]
Are those messages still with no plugins loaded?

**Comment by ginsterbusch** [comment]
No, sorry, those where actually the ones when I'm using Geany in regular mode. At this point, I cannot really do my job without the most essential plugins disabled (or else I'd have to switch to a different IDE, eg. Netbeans), which are as follows:

- shiftcolumn.so
- addons.so
- Project Organizer
- htmlchars.so
- lineoperations.so

After going through them for a bit, I decided to disable the following ones:

- GeanyVC - using the Terminal window / commandline for git all the time
- Commander
- Save Actions / saveactions.so - always been enabled, but no use for it in my current system setup; CTRL + S is being used unconsciously every few minutes anyway, no matter which program
- Code Navigation

gonna watch the debugging messages once in a while, too.

cu, w0lf.

**Comment by ginsterbusch** [comment]
I've also gotta point out, that I see this issue as a mere annoyance, not a critical bug. It'd be nice to have a workaround for this, even temporary, though.

I got powerful tools like a proper clipboard manager sitting at my fingertips, so if a variable / auto-replacement update sometimes doesnt go through (as its supposed to) or takes longer than expected, I just resort to classic copy + paste with a nifty 30 item history ;)

cu, w0lf.

cu, w0lf.
